### ISSUE TITLE
False positive UnusedVariable when using unset

### ISSUE THREAD

**Issue by kkmuffme** [vimeo/psalm#7711]
https://psalm.dev/r/10b22a9eab

Clearly not unused.
When I do "Fix code" it produces completely wrong code.

**Comment by orklah** [comment]
$array being unused makes $bar unused. When you use the array, it disappear: https://psalm.dev/r/f83d87c829.

However, the quick fix is indeed wrong.

**Comment by orklah** [comment]
Possibly, but I have no idea how it works at this level. I think it must treat each offset of the array separately in order to see that $bar don't get used at the end, but on the array level on the whole, there is both read access and write access. Not sure if the system can see this

**Comment by AndrolGenhald** [comment]
> I think it must treat each offset of the array separately in order to see that $bar don't get used at the end

Not exactly, `$bar` gets marked as used if `$array` is used at all, even if `$array['bar']` is not used: https://psalm.dev/r/c71aac6b06
